### PR TITLE
1193: set trial session for calendar validation

### PR DESCRIPTION
**Pull request by sutt0n**

### CODE REVIEW

File: shared/src/business/entities/trialSessions/TrialSession.test.js [hunk 1]
```diff
@@ -320,4 +320,42 @@ describe('TrialSession entity', () => {
...
+        },
+        {
+          applicationContext,
+        },
+      );
+
+      expect(trialSession.canSetAsCalendared()).toBeTruthy();
+    });
+
+    it('should NOT be able to set a trial session as calendared if one or more properties are not null', () => {
```
Comment (matthopson) [inline shared/src/business/entities/trialSessions/TrialSession.test.js]: Harnessing my inner Waldo here: `null` is not exactly represented in the test case. Perhaps "empty" is better (as described in the function's jsdocs)

Comment (matthopson) [reply shared/src/business/entities/trialSessions/TrialSession.test.js]: Same maybe applies to the others as well?

File: web-client/src/presenter/actions/TrialSession/canSetTrialSessionToCalendarAction.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { state } from 'cerebral';
+
+/**
+ * create a trial session
```
Comment (matthopson) [inline web-client/src/presenter/actions/TrialSession/canSetTrialSessionToCalendarAction.js]: Doesn't seem accurate

File: web-client/src/presenter/actions/TrialSession/canSetTrialSessionToCalendarAction.test.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      }),
+    };
+
+    presenter.providers.path = {
+      no: pathNoStub,
+      yes: pathYesStub,
+    };
+  });
+
+  it('has invalid criteria to set trial session as calendared', async () => {
```
Comment (matthopson) [inline web-client/src/presenter/actions/TrialSession/canSetTrialSessionToCalendarAction.test.js]: I don't know if this is even worth mentioning, but I think we usually specify the result in the test description... i.e. `Should return the no path when it has invalid criteria...`.

Same below... take it or leave it ¯\_(ツ)_/¯

**Review by matthopson — APPROVED** [review]
A couple of nits - otherwise 👍

File: web-client/src/presenter/actions/TrialSession/canSetTrialSessionToCalendarAction.test.js (hunk 3 above)
Comment (sutt0n) [reply web-client/src/presenter/actions/TrialSession/canSetTrialSessionToCalendarAction.test.js]: Yeah, I was initially going to do that, but didn't want to be too focused on the technical aspect within the unit test, but I'll change it for consistency's sake.

**Pull request merged**